src/pages/preferences/preferencesTabs/publicKeysTab/EditModal.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useState, useEffect } from "react";
+import { Modal } from "components/Modal";
+import { useMutation, useQuery } from "@apollo/react-hooks";
+import styled from "@emotion/styled";
+import Button, { Variant } from "@leafygreen-ui/button";
+import {
+  GetMyPublicKeysQuery,
+  UpdatePublicKeyMutation,
+  UpdatePublicKeyMutationVariables,
+  GetMyPublicKeysQueryVariables,
+  CreatePublicKeyMutation,
+  CreatePublicKeyMutationVariables,
+} from "gql/generated/types";
+import { GET_MY_PUBLIC_KEYS } from "gql/queries";
+import { UPDATE_PUBLIC_KEY } from "gql/mutations/update-public-key";
+import { useBannerDispatchContext } from "context/banners";
+import { Input } from "antd";
+import { CREATE_PUBLIC_KEY } from "gql/mutations/create-public-key";
+import { InputLabel, ErrorMessage } from "components/styles";
+import { v4 as uuid } from "uuid";
+
+const { TextArea } = Input;
+
+export interface EditModalPropsState {
+  replaceKey?: { name: string; key: string };
+  visible: boolean;
+}
+
+interface EditModalProps extends EditModalPropsState {
+  onCancel: () => void;
+}
+
+export const EditModal: React.FC<EditModalProps> = ({
+  replaceKey,
+  visible,
+  onCancel,
+}) => {
+  const { data: myKeysData } = useQuery<
+    GetMyPublicKeysQuery,
+    GetMyPublicKeysQueryVariables
+  >(GET_MY_PUBLIC_KEYS, { fetchPolicy: "cache-only" });
+  const dispatchBanner = useBannerDispatchContext();
+  const [errors, setErrors] = useState<string[]>([]);
+  const [updatePublicKey] = useMutation<
+    UpdatePublicKeyMutation,
+    UpdatePublicKeyMutationVariables
+  >(UPDATE_PUBLIC_KEY, {
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error editing the public key: ${error.message}`
+      );
+    },
+    onCompleted() {
+      dispatchBanner.clearAllBanners();
+    },
+    update(cache, { data }) {
+      cache.writeQuery<GetMyPublicKeysQuery, GetMyPublicKeysQueryVariables>({
+        query: GET_MY_PUBLIC_KEYS,
+        data: { myPublicKeys: [...data.updatePublicKey] },
+      });
+    },
+  });
+  const [createPublicKey] = useMutation<
+    CreatePublicKeyMutation,
+    CreatePublicKeyMutationVariables
+  >(CREATE_PUBLIC_KEY, {
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error creating the public key: ${error.message}`
+      );
+    },
+    onCompleted() {
+      dispatchBanner.clearAllBanners();
+    },
+    update(cache, { data }) {
+      cache.writeQuery<GetMyPublicKeysQuery, GetMyPublicKeysQueryVariables>({
+        query: GET_MY_PUBLIC_KEYS,
+        data: { myPublicKeys: [...data.createPublicKey] },
+      });
+    },
+  });
+
+  const [keyName, setKeyName] = useState<string>();
+  const [keyValue, setKeyValue] = useState<string>();
+
+  useEffect(() => {
+    setKeyName(replaceKey?.name ?? "");
```

Comment:
Won't doing this cause infinite rerenders? I think you need to use the initial state param above

Reply:
`replaceKey` is the initial state param and I can see how the name can be confusing. I followed the implementation I described in this comment: https://github.com/evergreen-ci/spruce/pull/352#discussion_r472251033
I can rename the key to `initialKey` or something to indicate that the prop represents an initial state